src/core/algorithms/nar/des/differential_functions.cpp

```diff
@@ -0,0 +1,58 @@
+#include "differential_functions.h"
+
+#include <algorithm>
+
+namespace algos::des {
+
+// gets slow if population ~= number_of_indices
+std::unordered_set<size_t> GetRandIndices(size_t except_index, size_t population,
+                                          size_t number_of_indices) {
+    std::unordered_set<size_t> indices;
+    indices.insert(except_index);
+    while (indices.size() < number_of_indices + 1) {
+        size_t random_index = (size_t)(RNG().Next() * population);
+        indices.insert(random_index);
+    }
+    indices.erase(except_index);
+    return indices;
```

Comment:
Don't understand the idea here. We want to generate unique indices (number_of_indices amount) ranging from 0 to  `population_size - 1`?
If so, it's a good idea to add an assert: `assert(number_of_indices <= population_size - 1)`

And, well, the current solution seems strange.
Maybe just:
- use `std::vector` of `population_size`
- fill it with values from 0 to `population_size - 1` using `std::iota`
- remove `except_index` with `indices.erase(indices.begin() + except_index);`
- shuffle indices with `std::shuffle`, using a random device from some class, or just create it here
- resize to `number_of_indices` with `indices.resize(number_of_indices);`.
- return `indices`
?

Reply:
`std::unordered_set` is used to prevent duplicate indices from appearing. I did consider the solution you are suggesting, but if `population_size` is 100 (usually it's bigger) and number_of_indices is 3 (doesn't get much bigger than that with most mutation strategies) it's a lot of shuffling just to discard 97 indices and leave the remaining 3. Your solution is good when `number_of_indices` is comparable to `population_size`, but with this algorithm it's usually the opposite, so my solution doesn't run into many "collisions" and works out ok.